src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -182,6 +191,79 @@ private void PopulateAssemblies()
             }
         }
 
+        /// <summary>
+        /// Update FileDefinitions and FileDependencies to pretend that certain Reference items
+        /// explicitly added by a .targets file in the NETStandard.Library are actually normal
+        /// package items. This allows them to show up properly under the "SDK" node in Solution
+        /// Explorer, rather than just under the "Assemblies" node.
+        ///
+        /// This is not meant to be a general mechanism for injecting files that can't be handled
+        /// through the normal NuGet package mechanisms.
+        /// </summary>
+        private void PopulatePreResolvedNetStandardAssemblies()
+        {
+            var preResolvedAssemblyDependencies = new List<ITaskItem>();
+            foreach (var reference in References)
+            {
+                var packageName = reference.GetMetadata("NuGetPackageId");
+                var packageVersion = reference.GetMetadata("NuGetPackageVersion");
+
+                // This is not a "pre-resolved" assembly; skip it.
+                if (packageName == null || packageVersion == null)
+                {
+                    continue;
+                }
+
+                // This is meant only as a workaround for an issue in NETStandard.Library 2.0.*.
+                if (!packageName.Equals("NETStandard.Library", StringComparison.OrdinalIgnoreCase))
```

Comment:
I don't agree with special casing here.  You already have the PackageIds that you are considering SDK references you should use those.